src/Affine.jl

```diff
@@ -0,0 +1,91 @@
+module AffineUtils
+
+using ..MLIR.API
+using ..MLIR.IR
+
+
+walk(f, other) = f(other)
+function walk(f, expr::Expr)
+    expr = f(expr)
+    Expr(expr.head, map(arg -> walk(f, arg), expr.args)...)
+end
+
+"""
+    @map (d1, d2, d3, ...)[s1, s2, ...] -> (d0 + d1, ...)
+
+Returns an affine map from the provided Julia expression.
+On the right hand side are allowed the following function calls:
+
+ - +, *, ÷, %, fld, cld
+
+The rhs can only contains dimensions and symbols present on the left hand side or integer literals.
+
+```juliadoctest
+julia> using MLIR: IR, AffineUtils
+
+julia> IR.context!(IR.Context()) do
+           AffineUtils.@map (d1, d2)[s0] -> (d1 + s0, d2 % 10)
+       end
+MLIR.IR.AffineMap(#= (d0, d1)[s0] -> (d0 + s0, d1 mod 10) =#)
+```
+"""
+macro map(ex)
```

Comment:
I would rename this to `affinemap`, to keep it consistent with MLIR and avoid confusion with Julia's `map`.
```suggestion
macro affinemap(ex)
```